exercises/pick/intro_to_qp.ipynb

```diff
@@ -0,0 +1,295 @@
+    "import os, sys\n",
+    "from urllib.request import urlretrieve\n",
+    "\n",
+    "if 'google.colab' in sys.modules and importlib.util.find_spec('manipulation') is None:\n",
+    "    urlretrieve(f\"http://manipulation.csail.mit.edu/scripts/setup/setup_manipulation_colab.py\",\n",
+    "                \"setup_manipulation_colab.py\")\n",
+    "    from setup_manipulation_colab import setup_manipulation\n",
+    "    setup_manipulation(manipulation_sha='dc0eb8aa32e271d9b5299a84435694d0565346af', drake_version='20200909', drake_build='nightly')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "tFMmTfbHWQfh"
+   },
+   "outputs": [],
+   "source": [
+    "# python libraries\n",
+    "import numpy as np\n",
+    "\n",
+    "from pydrake.all import (\n",
+    "    MathematicalProgram, Solve, eq, le, ge\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "ze9gQeOVOJUA"
+   },
+   "source": [
+    "# Introduction to MathematicalProgram \n",
+    "\n",
+    "The purpose of this exercise is to get you familiar with the basics of what an instance of an optimization problem is, as well as how to solve it. \n",
+    "\n",
+    "An optimization problem is usually written as \n",
+    "$$\\begin{aligned}\n",
+    "\\min_x \\quad & f(x) \\\\\n",
+    "\\textrm{s.t.} \\quad & g(x)\\leq 0,\\\\\n",
+    "\\quad &  h(x)=0 \\end{aligned}$$\n",
+    "\n",
+    "We call $x$ the **decision variable**, $f(x)$ the **cost function**, $g(x)\\leq 0$ an **inequality constraint**, and $h(x)=0$ an **equality constraint**. We usually denote the optimal solution by $x^*$. Most of the times, the constraints are hard-constraints, meaning that they must be fulfilled by the optimal solution. \n",
+    "\n",
+    "Drake offers a very good interface to many solvers using `MathematicalProgram`. Let's try to solve a simple problem using `MathematicalProgram`: \n",
+    "$$\\begin{aligned}\n",
+    "\\min_x \\quad & \\frac{1}{2}x^2 \\\\\n",
+    "\\textrm{s.t.} \\quad & x\\geq 3\n",
+    "\\end{aligned}$$\n",
+    "\n",
+    "Before we start coding, what do you expect the answer to be? You should persuade yourself that the optimal solution is $x^*=3$, since that is value at which minimum cost is achieved without violating the constraint.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "Khi7GeVNcwtU"
+   },
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "Steps to solve a optimization problem using Drake's MathematicalProgram\n",
+    "'''\n",
+    "# 1. Define an instance of MathematicalProgram \n",
+    "prog = MathematicalProgram() \n",
+    "\n",
+    "# 2. Add decision varaibles\n",
+    "x = prog.NewContinuousVariables(1)\n",
+    "\n",
+    "# 3. Add Cost function \n",
+    "prog.AddCost(x.dot(x))\n",
```

Comment:
BTW -- for the jacobians, it's actually nicer / easier to use `AddQuadraticErrorCost`.  i wonder if you should introduce that here? (or the diff IK problem)?